### ISSUE TITLE
[Bug] extra divs are generated around custom components for node

### ISSUE THREAD

**Issue by sytolk** [Saul-Mirone/milkdown#146]
**Describe the bug**
I heve use custom component for link node with react:
https://github.com/Saul-Mirone/milkdown/blob/main/gh-pages/pages/integrations/react/index.md#custom-component-for-node

```javascript
const TSLink: React.FC = ({ children }) => {
    const { node } = useNodeCtx();

    const clickLink = (evt: any) => {
      evt.preventDefault();

      window.parent.postMessage(
        JSON.stringify({
          command: 'openLinkExternally',
          link: node.attrs.href,
        }),
        '*'
      );
    };
    return (
      <a href="#" onClick={clickLink}>
        {children}
      </a>
    );
  };

const editor = useEditor(
    (root, renderReact) => {
      const nodes = commonmark
        .configure(paragraph, { view: renderReact(TSParagraph) })
        .configure(link, { view: renderReact(TSLink) })

      return (
        Editor.make()
          .config((ctx) => {
            context = ctx;
            ctx.set(rootCtx, root);
          })
          .use(nord)
          .use(nodes)
          .use(commonmark)
      );
    }
  );
```
but I see in DOM its have extra `<div>` around link:

My markdown is: `[github](http://github.com) after link inline text`

```html
<p class="paragraph css-5p70de">
<div>
<a href="#"><div><div>github</div></div></a>
</div> after link inline text
</p>
```
as the result " after link inline text" is displayed on the next line..

I have try to fix this and override paragraph component with `inline-flex` style
```javascript
 const TSParagraph: React.FC = ({ children }) => (
    <p>
      <div style={{ display: 'inline-flex' }}>{children}</div>
    </p>
  );
```
but there is extra `<div>` generated around paragraph node too:

```html
<div class="">
<p>
<div style="display: inline-flex;">
<div><div><div>
<a href="#" class="ts-link"><div><div>github</div></div></a>
</div> after link inline text</div></div></div>
</p>
</div>
```

**Comment by Saul-Mirone** [comment]
I think additional `div`s can't avoid due to the limitation of prosemirror's node view feature, however, I certainly can do some optimize to wrap the inline nodes with `span` instead of `div` and add some class name to make it easier to add style.

**Comment by keellyp** [comment]
Hey, I'm facing the same issue, and also when I'm trying to configure a custom link with 
```ts
.configure(link, { view: renderReact(CustomLink) })
```

I get this errror:
```ts
Type 'NodeViewFactory' is not assignable to type 'MarkViewFactory'
```

Do you know how I can fix it properly? :) 
Thanks for your help!

**Comment by Saul-Mirone** [comment]
> Hey, I'm facing the same issue, and also when I'm trying to configure a custom link with
> 
> ```ts
> .configure(link, { view: renderReact(CustomLink) })
> ```
> 
> I get this errror:
> 
> ```ts
> Type 'NodeViewFactory' is not assignable to type 'MarkViewFactory'
> ```
> 
> Do you know how I can fix it properly? :) Thanks for your help!

It's a bug and I've already fixed it in main branch, will be released in this week.

**Comment by keellyp** [comment]
That's great, thanks for your quick answer! 🙏 
Also, do you know when the fix for generated divs around custom components will be available?

**Comment by Saul-Mirone** [comment]
> That's great, thanks for your quick answer! 🙏 
> 
> Also, do you know when the fix for generated divs around custom components will be available? 

They are all done in main branch now. Div are still there, however, it will be optimized as mentioned above.

**Comment by Saul-Mirone** [comment]
Released in v4.13
